micropip/_compat_in_pyodide.py

```diff
@@ -37,10 +36,7 @@ async def fetch_string_and_headers(
     response = await pyfetch(url, **kwargs)
 
     content = await response.string()
-    # TODO: replace with response.headers when pyodide>= 0.24 is released
-    headers: dict[str, str] = Object.fromEntries(
-        response.js_response.headers.entries()
-    ).to_py()
```

Comment:
So it's a bit of a question how fast we drop support for older pyodide releases in micropip.

I'm not opposed to doing this but just something I wonder about, especially that here there is no major benefits from updating (aside from having slightly simpler code in this case).